spring-integration-redis/src/test/java/org/springframework/integration/redis/outbound/ReactiveRedisStreamMessageHandlerTests.java

```diff
@@ -0,0 +1,223 @@
+import org.springframework.data.redis.connection.ReactiveRedisConnectionFactory;
+import org.springframework.data.redis.connection.stream.ObjectRecord;
+import org.springframework.data.redis.connection.stream.StreamOffset;
+import org.springframework.data.redis.core.ReactiveRedisTemplate;
+import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer;
+import org.springframework.data.redis.serializer.RedisSerializationContext;
+import org.springframework.data.redis.serializer.RedisSerializer;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+import org.springframework.integration.handler.ReactiveMessageHandlerAdapter;
+import org.springframework.integration.redis.rules.RedisAvailable;
+import org.springframework.integration.redis.rules.RedisAvailableTests;
+import org.springframework.integration.redis.store.RedisMessageStoreTests;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+/**
+ * @author Attoumane Ahamadi
+ *
+ * @since 5.4
+ */
+@ContextConfiguration
+@RunWith(SpringRunner.class)
+@DirtiesContext
+@SuppressWarnings({"unchecked", "rawtypes"})
+public class ReactiveRedisStreamMessageHandlerTests extends RedisAvailableTests {
+
+	@Autowired
+	@Qualifier("forRedisStreamChannel")
+	private MessageChannel messageChannel;
+
+	@Autowired
+	private ReactiveRedisConnectionFactory redisConnectionFactory;
+
+	@Autowired
+	private ReactiveMessageHandlerAdapter handlerAdapter;
+
+	@Test
+	@RedisAvailable
+	public void emptyStreamKeyTest() {
+		assertThatIllegalArgumentException()
+				.isThrownBy(() -> new ReactiveRedisStreamMessageHandler("", null));
+	}
+
+	@Test
+	@RedisAvailable
+	public void nullConnectionFactoryTest() {
+		assertThatIllegalArgumentException()
+				.isThrownBy(() -> new ReactiveRedisStreamMessageHandler("stream", null));
+	}
+
+
+	@Test
+	@RedisAvailable
+	public void simpleStringInsertionTest() {
+		String streamKey = "myStream";
+		String messagePayload = "Bonjour à tous les confinés";
+
+		handlerAdapter.handleMessage(new GenericMessage<>(messagePayload));
+
+		RedisSerializationContext<String, String> serializationContext = redisStringOrJsonSerializationContext(true, null);
+
+		ReactiveRedisTemplate<String, String> template = new ReactiveRedisTemplate(redisConnectionFactory, serializationContext);
+
+		ObjectRecord<String, String> record = template.opsForStream().read(String.class, StreamOffset.fromStart(streamKey)).blockFirst();
+		assertThat(record.getStream()).isEqualTo(streamKey);
+		assertThat(record.getValue()).isEqualTo(messagePayload);
+		template.delete(streamKey).block();
+	}
+
+	@Test
+	@RedisAvailable
+	public void integrationStreamOutboundTest() {
```

Comment:
I don't see difference between this and previous test...

Reply:
This one use a `MessageChannel`